frame/nis/src/lib.rs

```diff
@@ -689,13 +708,50 @@ pub mod pallet {
 			receipt.proportion.saturating_reduce(proportion);
 			summary.proportion_owed.saturating_reduce(proportion);
 
-			T::Currency::transfer(&our_account, &who, amount, AllowDeath)
-				.map_err(|_| Error::<T>::Unfunded)?;
-
 			let dropped = receipt.proportion.is_zero();
+
+			if amount > on_hold {
+				T::Currency::unreserve_named(&T::ReserveId::get(), &who, on_hold);
+				let deficit = amount - on_hold;
+				// Try to transfer deficit from pot to receipt owner.
+				summary.receipts_on_hold -= on_hold;
```

Comment:
Sounds like it should really use saturating math here:

```suggestion
				summary.receipts_on_hold = summary.receipts_on_hold.saturating_sub(on_hold);
```

Reply:
Or better yet:

```suggestion
				summary.receipts_on_hold.saturating_reduce(on_hold);
```